crates/cheatcodes/spec/src/vm.rs

```diff
@@ -1531,6 +1531,10 @@ interface Vm {
     #[cheatcode(group = Filesystem)]
     function writeLine(string calldata path, string calldata data) external;
 
+    /// Gets the artifact path from the creation code.
+    #[cheatcode(group = Filesystem)]
+    function getArtifactPath(bytes calldata creationCode) external view returns (string memory path);
```

Comment:
imo it's not obvious what this function accepts as argument. should we make this more explicit, like getArtifactPathByCode or smth @grandizzy ?

Reply:
agree that `getArtifactPathByCode` is cleaner, also since for `getCode` we have `getCode(artifactPath)` and `getDeployedCode(artifactPath)` maybe we need to add a `getArtifactPathByDeployedCode` too? (where the only difference is that `ContractsByArtifact.find_by_deployed_code` instead `find_by_creation_code` is called)

Reply:
+1 to both. 

Makes sense to name them explicitly, also avoids signature conflicts.